alacritty/src/config/bindings.rs

```diff
@@ -734,23 +628,83 @@ pub fn platform_key_bindings() -> Vec<KeyBinding> {
     vec![]
 }
 
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
-pub enum Key {
-    Scancode(u32),
-    Keycode(VirtualKeyCode),
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum BindingKey {
+    Scancode(KeyCode),
+    Keycode { key: Key, location: KeyLocation },
 }
 
-impl<'a> Deserialize<'a> for Key {
+impl<'a> Deserialize<'a> for BindingKey {
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where
         D: Deserializer<'a>,
     {
         let value = SerdeValue::deserialize(deserializer)?;
         match u32::deserialize(value.clone()) {
-            Ok(scancode) => Ok(Key::Scancode(scancode)),
+            Ok(scancode) => Ok(BindingKey::Scancode(KeyCode::from_scancode(scancode))),
             Err(_) => {
-                let keycode = VirtualKeyCode::deserialize(value).map_err(D::Error::custom)?;
-                Ok(Key::Keycode(keycode))
+                let keycode = String::deserialize(value.clone()).map_err(D::Error::custom)?;
+                let (key, location) = if keycode.len() == 1 {
+                    (Key::Character(keycode.to_lowercase().into()), KeyLocation::Standard)
+                } else {
```

Comment:
Why are you checking for `keycode.len() == 1` here? Shouldn't it be based on the number of characters not bytes?